sql/mozfun/glam/map_from_array_offsets_precise/udf.sql

```diff
@@ -0,0 +1,45 @@
+-- udf_get_values
+CREATE OR REPLACE FUNCTION glam.map_from_array_offsets_precise(
+  required ARRAY<FLOAT64>,
+  `values` ARRAY<FLOAT64>
+)
+RETURNS ARRAY<STRUCT<key STRING, value FLOAT64>> AS (
+  (
+    SELECT
+      ARRAY_AGG(
+        STRUCT<key STRING, value FLOAT64>(
+          CAST(ROUND((key / 10), 1) AS STRING),
+          `values`[OFFSET(SAFE_CAST(key AS INT64))]
+        )
+        ORDER BY
+          key
+      )
+    FROM
+      UNNEST(required) AS key
+  )
+);
+
+SELECT
+  assert.array_equals(
+    ARRAY<STRUCT<key STRING, value FLOAT64>>[("0", 2.0), ("0.2", 4.0)],
+    glam.map_from_array_offsets_precise([0.0, 2.0], [2.0, 3.0, 4.0])
+  ),
+  -- required ordered
+  assert.array_equals(
+    ARRAY<STRUCT<key STRING, value FLOAT64>>[("0", 2.0), ("0.2", 4.0)],
+    glam.map_from_array_offsets_precise([2.0, 0.0], [2.0, 3.0, 4.0])
+  ),
+  -- intended use-case, for approx_quantiles
+  assert.array_equals(
+    ARRAY<STRUCT<key STRING, value FLOAT64>>[("25", 1.0), ("50", 2.0), ("75", 3.0), ("99.9", 4.0)],
+    glam.map_from_array_offsets_precise(
+      [250.0, 500.0, 750.0, 999.0],
```

Comment:
Can we change this to `["25.0", "50.0", "75.0", "99.9"]` and then above on line 12 it can be `key * 10` and line 11 doesn't need to divide by 10.

I think this will be just more clear that the input to the function is the required percentiles.